scratchpad/bench/base.py

```diff
@@ -57,35 +60,35 @@ def gen_tensors(cls, num_tensors, *sizes, dtype='float'):
             tensors.append(tensor)
         return tensors
     
-    @classmethod
-    def gen_tensor(cls, *sizes, dtype='float'):
-        raise NotImplementedError
-    
-    @classmethod
-    def get_ready(cls, num_tensors, *sizes):
-        return num_tensors, *sizes
+
+
+class Benchmark:
+    timing=pyrofiler.timing
 
     @staticmethod
-    def get_operation():
+    def get_task_type():
         raise NotImplementedError
-        
 
     @classmethod
-    def benchmark(cls, num_tensors, *sizes, dtype='float', **args):
-        num_tensors, *sizes = cls.get_ready(num_tensors, *sizes)
+    def get_operation(cls):
+        raise NotImplementedError
+        
+    @classmethod
+    def benchmark(cls, backend:Backend, num_tensors, *sizes, dtype='float', **args):
+        num_tensors, *sizes = backend.get_ready(num_tensors, *sizes)
         operation = cls.get_operation()
```

Comment:
Here's a suggestion: this method sholud `raise NotImplementedError`, and then you'll have two children classes `class MatmulBench(Benchmark` and `class TNcontractBench(Benchmark)`. The children classes will take the operation directly from `backend.matmul` or `backend.tncontract`. This way you can handle different operations' API in the benchmark() method